src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml.cpp

```diff
@@ -114,9 +119,13 @@ void EquationInputArea::EquationTextBox_RemoveButtonClicked(Object ^ sender, Rou
 
 void EquationInputArea::EquationTextBox_KeyGraphFeaturesButtonClicked(Object ^ sender, RoutedEventArgs ^ e)
 {
+    // ensure the equation has been submitted before trying to get key graph features out of it
+    EquationInputArea::InputTextBox_Submitted(sender, e);
```

Comment:
I think this would degrade the KGF analysis performance since we have to wait for the graph again, is this necessary? I think it would still work without this and also graph if it hasn't already due to losing focus.

Reply:
So I really wanted to have the experience be "Enter the equation, hit the context menu key, and then hit enter".  Without triggering submitted, it would not actually update the method, at which point the eq object wouldn't have the right equation.  Because the context menu is on the rich edit, focus isn't leaving the rich edit and thus the blur code is not being run...  Agreed about the perf implications though.  I'm not entirely sure on what the right thing to do is though.

Reply:
added a check to see if we have focus.  If we do, then we submit otherwise we don't.  I think that this avoids calling the graph in cases where we know the graph is up to date.

Reply:
Good idea!